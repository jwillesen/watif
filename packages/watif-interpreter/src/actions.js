import Immutable from 'immutable'
import { createActions } from 'redux-actions'

export const {
  setItemState,
  setCurrentRoom,
  setCurentItem,
  computeDisplayValues,
  executeVerb,

} = createActions({

/* eslint-disable indent */
SET_ITEM_STATE: (itemId, newItemState) => { // (string, [Object | Immutable.Map])
  if (!Immutable.Map.isMap(newItemState)) {
    newItemState = Immutable.fromJS(newItemState)
  }
  return { itemId, newItemState }
},

},

'SET_CURRENT_ROOM',       // (itemId: string)
'SET_CURRENT_ITEM',       // (itemId: string)
'ADD_LOG_ENTRY',          // (entry: watext)

'EXECUTE_VERB',           // (verb invokation)

'COMPUTE_DISPLAY_VALUES'  // ()

)
/* eslint-enable indent */

// verb invokation:
//  {
//    verb: string,
//    subject: itemId,
//    object: itemId,
//  }
