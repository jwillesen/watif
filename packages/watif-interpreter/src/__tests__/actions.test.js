import * as matchers from 'jest-immutable-matchers'
import * as Actions from '../actions'
import Immutable from 'immutable'

beforeAll(() => {
  jest.addMatchers(matchers)
})

describe('setItemState', () => {
  it('accepts a Map', () => {
    const newStateMap = Immutable.Map({foo: 'bar'})
    const result = Actions.setItemState('42', newStateMap)
    expect(result).toMatchSnapshot()
  })

  it('deeply converts a pojo to Immutable.Map', () => {
    const newStatePojo = { foo: 'bar', baz: {bing: 42} }
    const result = Actions.setItemState('42', newStatePojo)
    const newItemState = result.payload.newItemState
    expect(Immutable.Map.isMap(newItemState)).toBeTruthy()
    expect(Immutable.Map.isMap(newItemState.get('baz'))).toBeTruthy()
  })
})
