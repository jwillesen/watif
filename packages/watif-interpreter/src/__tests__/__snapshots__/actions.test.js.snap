// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setItemState accepts a Map 1`] = `
Object {
  "payload": Object {
    "itemId": "42",
    "newItemState": Immutable.Map {
      foo: "bar",
},
  },
  "type": "SET_ITEM_STATE",
}
`;
